File: nth_fibonacci/✅ nth_fib_memoize.py
def getNthFib(n, 
    memoize = {
        1: 0, 
        2: 1
    }
):
    ## -- ##
    if n in memoize:
        return memoize[n]
    else: 
        memoize[n] = getNthFib(n - 1, memoize) + getNthFib(n - 2, memoize)
        return memoize[n]


print(getNthFib(2))
print(getNthFib(5))
print(getNthFib(10))
print(getNthFib(50))

File: ⭐️ Favorites ⭐️/random/⭐️ nth_fibonacci/✅ nth_fib_memoize.js
// TimeComplexity = O(n) => Only need to calculate value once for each `n`
// SpaceComplexity = O(n) => Space on the call stack

function nth_fib_memoize(n) {
    // Data
    memoize = {
        1: 0,
        2: 1
    }
    // Logic
    if (memoize.includes(n)) { return memoize[n] } 
    else {
        memoize[n] =  nth_fib_memoize(n - 1) + nth_fib(n - 2)
        return memoize[n]
    }
}

console.log(nth_fib(6))